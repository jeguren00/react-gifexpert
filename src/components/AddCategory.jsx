import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({onAddCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) {
            return;
        }
        onAddCategory(inputValue.trim());
        setInputValue('');
    }
    return (
        <form aria-label="form" onSubmit={onSubmit}>
            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
             />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
  }