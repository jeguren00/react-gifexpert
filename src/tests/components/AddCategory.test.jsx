import { fireEvent, render,screen } from "@testing-library/react"
import { AddCategory } from "../../components";

describe('pruebas con componente addcategory', () => {
    const inputValue = "fallout";

    test('match con el snapshot', () => { 
        const onAddCategory = jest.fn();
        const {container} = render(<AddCategory onAddCategory={onAddCategory} />);

        expect(container).toMatchSnapshot();
     });


    test('debe de cambiar del input', () => { 
        render(<AddCategory onAddCategory={() => {}} />);
        const input = screen.getByRole('textbox');
        fireEvent.input(input, {target: {value: inputValue}});

        expect(input.value).toBe(inputValue);
     });

     test('debe de llamar onAddCategory si el input recibe un valor ', () => {
        const onAddCategory = jest.fn();
        render(<AddCategory onAddCategory={onAddCategory} />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, {target: {value: inputValue}});
        fireEvent.submit(form);
        expect(input.value).toBe('');
        expect(onAddCategory).toHaveBeenCalled();
        expect(onAddCategory).toHaveBeenCalledTimes(1);
        expect(onAddCategory).toHaveBeenCalledWith(inputValue);
     });

     test('no debe de llamar la funcion si el input esta vacio', () => { 
        const onAddCategory = jest.fn();
        render(<AddCategory onAddCategory={onAddCategory} />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, {target: {value: ''}});
        fireEvent.submit(form);
        expect(onAddCategory).toHaveBeenCalledTimes(0);
      });

 })