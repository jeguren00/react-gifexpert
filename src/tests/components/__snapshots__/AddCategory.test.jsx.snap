// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con componente addcategory match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
